authorizer/src/utils/auth_utils.py

```diff
@@ -0,0 +1,168 @@
+    principal: Principal, 
+    expected_group_ids: List[str]
+) -> List[str]:
+    """Performs the authz validation from principal and expected group ids.
+
+    Parameters
+    ----------
+    identity_client: ExtendedIdentityDataPlaneClient
+        An instance of ExtendedIdentityDataPlaneClient.
+    principal: Principal
+        An instance of Principal.
+    expected_group_ids: list
+        A list of allowed group ids.
+
+    Returns
+    -------
+    List:
+        A list of allowed group ids.
+    """
+    filter_group_membership_details = FilterGroupMembershipDetails()
+    filter_group_membership_details.principal = principal
+    filter_group_membership_details.group_ids = expected_group_ids
+    membership_response = identity_client.filter_group_membership(filter_group_membership_details)
+    membership_result: FilterGroupMembershipResult = membership_response.data
+    if not set(expected_group_ids).intersection(membership_result.group_ids):
+        raise AuthorizationException(membership_response.status, expected_group_ids, principal.subject_id)
+    return membership_result.group_ids
+
+
+def get_group_ids_from_config(config: Dict) -> List[str]:
+    """Gets group ids from config.
+
+    Parameters
+    ----------
+    config: dict
+        A dict of configurations
+
+    Returns
+    -------
+    List
+        A list of group ids seperated in the original string by ','
+    """
+    group_ids = config.get(ALLOWED_GROUP_IDS)
+    group_ids = group_ids.replace(" ", "")
+    return group_ids.split(',')
+
+
+def _get_internal_instance_principal_signer() -> InstancePrincipalsSecurityTokenSigner:
+    """Overrides metadata url of InstancePrincipalSigner class"""
+    override_metadata_url = os.getenv(
+        "METADATA_OVERRIDE_URL",
+        InstancePrincipalsSecurityTokenSigner.METADATA_URL_BASE
+    )
+    InstancePrincipalsSecurityTokenSigner.METADATA_URL_BASE = override_metadata_url
+    InstancePrincipalsSecurityTokenSigner.GET_REGION_URL = \
+        '{}/instance/region'.format(override_metadata_url)
+    InstancePrincipalsSecurityTokenSigner.LEAF_CERTIFICATE_URL = \
+        '{}/identity/cert.pem'.format(override_metadata_url)
+    InstancePrincipalsSecurityTokenSigner.LEAF_CERTIFICATE_PRIVATE_KEY_URL = \
+        '{}/identity/key.pem'.format(override_metadata_url)
+    InstancePrincipalsSecurityTokenSigner.INTERMEDIATE_CERTIFICATE_URL = \
+        '{}/identity/intermediate.pem'.format(override_metadata_url)
+    return InstancePrincipalsSecurityTokenSigner()
+
+
+def _get_env_bool(env_var: str, default: bool = False) -> bool:
+    env_val = os.getenv(env_var)
+    if env_val is None:
+        env_val = default
+    else:
+        env_val = env_val.lower()
+        if env_val == "true":
+            env_val = True
+        elif env_val == "false":
+            env_val = False
+        else:
+            raise ValueError("For environment variable: {0} only string values T/true or F/false are allowed but: \
```

Comment:
The error message says that T and F also acceptable, but I don't see this logic.

I guess we could do something like this

```
env_val = os.getenv(env_var)

if env_val is None:
   return default

return env_val.lower() in ("true","t","1")
```

Reply:
Thank you for noticing this, just updated